src/api/v1beta1/feature_flags.go

```diff
@@ -115,33 +111,43 @@ func (dk *DynaKube) getDefaultIgnoredNamespaces() []string {
 // FeatureAutomaticKubernetesApiMonitoring is a feature flag to enable automatic kubernetes api monitoring,
 // which ensures that settings for this kubernetes cluster exist in Dynatrace
 func (dk *DynaKube) FeatureAutomaticKubernetesApiMonitoring() bool {
-	return dk.Annotations[annotationFeatureAutomaticKubernetesApiMonitoring] == "true"
+	return dk.getFeatureFlagRaw(annotationFeatureAutomaticKubernetesApiMonitoring) == "true"
 }
 
 // FeatureDisableMetadataEnrichment is a feature flag to disable metadata enrichment,
 func (dk *DynaKube) FeatureDisableMetadataEnrichment() bool {
-	return dk.Annotations[annotationFeatureDisableMetadataEnrichment] == "true"
+	return dk.getFeatureFlagRaw(annotationFeatureDisableMetadataEnrichment) == "true"
 }
 
 // FeatureUseActiveGateImageForStatsd is a feature flag that makes the operator use ActiveGate image when initializing Extension Controller and Statsd containers
 // (using special predefined entry points).
 func (dk *DynaKube) FeatureUseActiveGateImageForStatsd() bool {
-	return dk.Annotations[annotationFeatureUseActiveGateImageForStatsd] == "true"
+	return dk.getFeatureFlagRaw(annotationFeatureUseActiveGateImageForStatsd) == "true"
 }
 
 // FeatureCustomEecImage is a feature flag to specify custom Extension Controller Docker image path
 func (dk *DynaKube) FeatureCustomEecImage() string {
-	return dk.Annotations[annotationFeatureCustomEecImage]
+	return dk.getFeatureFlagRaw(annotationFeatureCustomEecImage)
 }
 
 // FeatureCustomStatsdImage is a feature flag to specify custom StatsD Docker image path
 func (dk *DynaKube) FeatureCustomStatsdImage() string {
-	return dk.Annotations[annotationFeatureCustomStatsdImage]
+	return dk.getFeatureFlagRaw(annotationFeatureCustomStatsdImage)
 }
 
 // FeatureDisableReadOnlyOneAgent is a feature flag to specify if the operator needs to deploy the oneagents in a readonly mode,
 // where the csi-driver would provide the volume for logs and such
 // Defaults to false
 func (dk *DynaKube) FeatureDisableReadOnlyOneAgent() bool {
-	return dk.Annotations[AnnotationFeatureDisableReadOnlyOneAgent] == "true"
+	return dk.getFeatureFlagRaw(AnnotationFeatureDisableReadOnlyOneAgent) == "true"
+}
+
+func (dk *DynaKube) getFeatureFlagRaw(annotation string) string {
```

Comment:
Consider moving it to `internal_operator_flags.go`

Reply:
I feel it wouldnt be in a better place in `internal_operator_flags.go` as that file is for internal flags, and yes this function is a more general function, but I think its fine here :D

Reply:
OK. IMO `getFeatureFlagRaw()` and functions for internal flags belong together but some further refactoring would be needed, not mandatory at this point.